packages/espressocash_app/lib/features/ramp/partners/coinflow/widgets/launch.dart

```diff
@@ -100,6 +114,40 @@ extension BuildContextExt on BuildContext {
     await WebViewRoute((url: blank, onLoaded: handleLoaded, title: null))
         .push<void>(this);
   }
+
+  Future<bool> _hasKYC({required String address}) =>
+      runWithLoader<bool>(this, () async {
+        try {
+          final client = sl<CoinflowClient>();
+
+          await client.getWithdrawer(address);
+
+          return true;
+        } on Exception {
```

Comment:
Catching any exception is not reliable, it can be just a network outage. Can it be more specific, does their API return some explicit error?

Reply:
Their API returns non 200 status code if the user doesn't have kyc [(ref)](https://docs.coinflow.cash/reference/getwithdrawer). 

I've updated the code to catch their specific status codes, and show error toast for anyting else